### ISSUE TITLE
[APP V2] Problem with button

### ISSUE THREAD

**Issue by daneedk** [athombv/homey#2092]
* **Homey Version**: 1.5.6-rc7
* **Homey Log Code**: N/A
* **Smartphone Platform**: iPhone
* **Smartphone App Version**: 1.1.105

**If related to a Homey App:**
* **Homey App Version**:
* **Homey App Log Code**:
* **Affected Devices (preferably product codes)**:

**What did you try to achieve?**
Use a button

**What did you expect as result?**
The button to function as a on/off switch

**What was the result?**
The button changed state but immediately switched back

**Any other remarks**
It seems this happens to all buttons with a class 'other' that have the 'onoff' capability.

The button is shown like this in the old app:
![image](https://user-images.githubusercontent.com/16610696/34198365-6d8478b6-e56a-11e7-97aa-804c9b5dc92e.png)
and it functions as described in the developer documentation, you can either click the image or the on/off slider. When the button is off the image it's greyed out.

The button is shown like this in the web interface:
![image](https://user-images.githubusercontent.com/16610696/34198437-af840362-e56a-11e7-9399-1586db56c7f3.png)
And it functions the same as described above.

In the new app the button is shown like this:
![image](https://user-images.githubusercontent.com/16610696/34198530-fe9f2918-e56a-11e7-8df0-00ad95bf15bf.png)
Notice the missing on/off slider.
When you open the button it looks like this:
![image](https://user-images.githubusercontent.com/16610696/34198569-2e19f40c-e56b-11e7-8d1b-7ec496929c9b.png)
When the state is changed in this view it switches straight back to the previous state, while still triggering the action that has to be done on the first change, So when going from off to on, the action that 'on' triggers is run, but the action that 'off' triggers is not run, while the interface will show the 'off' state.

On a previous app version this did work, unfortunately I cannot say how many versions back this was.

**Comment by bvdbos** [comment]
You mean a previous version of the experimental Homey mobile app? Does this behaviour also happen for other Homey-apps like KaKu? In other words, is this issue specific for the Surveillance-app?

**Comment by daneedk** [comment]
Yes, it worked correctly on a previous version of the Homey mobile app.
It's not specific for the Surveillance app, other apps that have buttons (devices) that have class 'other' and the 'onoff' capability show the same behavious, for example a mode button created with the Virtual Devices app. It can very well be that other class/capability combinations show the same behaviour but I've not tested for that.

**Comment by zidodiz** [comment]
Experiencing the same problem. I have a virtual device that is a mode changing device and it shows the exact same problem as described above. In preview app in Android the switch is not shown on the overview page of devices. In the device itself there is a on/off slide but sliding it only makes it slide back automatically. However when it has been set to the one mode it is actually set to that mode even if the GUI changes it back directly after setting it. This can be seen by leaving the device page and enter it again and then the device will have the slider set to the expected value even though it was "reset" when it was set. Same scenario in both directions on to off or off to on makes no difference the GUI resets the value that the user wants to set.
